mirrord/layer/src/lib.rs

```diff
@@ -195,8 +195,17 @@ fn layer_pre_initialization() -> Result<(), LayerError> {
 }
 
 /// Initialize a new session with the internal proxy.
-/// Sets [`PROXY_CONNECTION`].
+/// Sets [`PROXY_CONNECTION`] if not in trace only mode.
 fn load_only_layer_start(config: &LayerConfig) {
```

Comment:
This is called from the layer constructor when we load to a skipped process, e.g. compiler.
Proxy connection is made here to prevent issues with long builds